examples/02-plot/vector-component.py

```diff
@@ -0,0 +1,44 @@
+"""
+Vector Component
+~~~~~~~~~~~~~~~~
+
+Plot a single component of a vector as a scalar array.
+
+We can plot individual components of multi-component arrays with the
+``component`` argument  of the ``add_mesh`` method.
+"""
+
+import numpy as np
+import pyvista as pv
+
+###############################################################################
+# Start with a plane containing a vector with value of [1, 2, 3]
+# everywhere. The magnitude of this vector is 3.74.
+mesh = pv.Plane()
+data = np.ones([mesh.n_cells, 3])
+data[:] = [1.0, 2.0, 3.0]
+mesh["Velocity Vector"] = data
+###############################################################################
+# The default behavior with no component specified is to use the vector
+# magnitude. We can access each component by specifying the component argument.
+dargs = dict(
+    scalars="Velocity Vector",
+    cmap="rainbow",
+    clim=[1.0, 4.0],
+    scalar_bar_args = dict(n_labels=4)
+)
+p = pv.Plotter(shape=(2,2))
+p.subplot(0,0)
+p.add_mesh(mesh, **dargs)
+p.add_text("Vector Magnitude")
+p.subplot(0,1)
+p.add_mesh(mesh.copy(), component=0, **dargs)
```

Comment:
Is it important to call out why `mesh.copy()` is required here?  Since this duplicates a potentially large dataset in it's entirety, does it also make sense to also provide an alternative method in the example like:

```
mesh["Velocity Vector_0"] = mesh["Velocity Vector"][0]
p.add_mesh(mesh, scalars="Velocity Vector_0")
```

Reply:
I'm not actually sure if there is a memory advantage using either approach, which is why I'm asking for feedback.

Reply:
By my testing with a large dataset, using a copy of the mesh for each component for plotting consumes ~3x more memory than just creating a copy of each component in the mesh data.  I think there is still a good use case for the use in this PR when only a single component is plotted, or when the dataset is small.  In the other case, I think can be addressed by adding it to the example.

<details><summary>code for obtaining memory usage</summary>
<p>

I used memory_profiler, and compared the total memory usage after `p.show()`.  Need to run twice commenting out one function call then the other.

`run` took 4150 MiB and `run2` took 1469 MiB.

```python
import numpy as np
import pyvista as pv
from memory_profiler import profile

n=3000

@profile
def run():
    mesh = pv.Plane(i_resolution=n, j_resolution=n)
    data = np.ones([mesh.n_cells, 3])
    data[:] = [1.0, 2.0, 3.0]
    mesh["Velocity Vector"] = data

    dargs = dict(
        scalars="Velocity Vector",
    )
    p = pv.Plotter(shape=(2,2))
    p.subplot(0,0)
    p.add_mesh(mesh, **dargs)
    p.subplot(0,1)
    p.add_mesh(mesh.copy(), component=0, **dargs)
    p.subplot(1,0)
    p.add_mesh(mesh.copy(), component=1, **dargs)
    p.subplot(1,1)
    p.add_mesh(mesh.copy(), component=2, **dargs)

    p.show()

@profile
def run2():
    mesh = pv.Plane(i_resolution=n, j_resolution=n)
    data = np.ones([mesh.n_cells, 3])
    data[:] = [1.0, 2.0, 3.0]
    mesh["Velocity Vector"] = data

    dargs = dict()
    p = pv.Plotter(shape=(2,2))
    p.subplot(0,0)
    p.add_mesh(mesh, scalars="Velocity Vector", **dargs)

    p.subplot(0,1)
    mesh["C0"] = mesh["Velocity Vector"][:,0]
    p.add_mesh(mesh, scalars="C0", **dargs)

    p.subplot(1,0)
    mesh["C1"] = mesh["Velocity Vector"][:,1]
    p.add_mesh(mesh, scalars="C1", **dargs)

    p.subplot(1,1)
    mesh["C2"] = mesh["Velocity Vector"][:,2]
    p.add_mesh(mesh, scalars="C2", **dargs)

    p.show()

#run()
run2()
```

</p>
</details>

Reply:
Issue with copying the mesh can be dealt with in a later PR.  This PR has already taken months to get through and I'm all for fixing this at a later date, especially as this will require a through refactor of `plotter.py`